metabase/metabase
# Delete unused new question route

Closes #19677

Here we clean up the route and page where we used to display a question type picker (simple question / custom question / native query).

We update code for the model equivalent so that it reflects the fact that this directory is now all about models.

Also, we s/dataset/model everywhere we touch here to converge our codebase towards calling models models.

### How to Test

Go to http://localhost:3000/question/new

You should no longer see the question type picker, instead land on the data picker for a custom question.

The app hasn't had a link to `/question/new` since these links are now inside `+ New`.
